proctor-common/src/main/java/com/indeed/proctor/common/model/Payload.java

```diff
@@ -326,7 +326,42 @@ public boolean equals(final Object o) {
                 Objects.equals(stringValue, payload.stringValue) &&
                 Arrays.equals(stringArray, payload.stringArray) &&
                 Objects.equals(json, payload.json) &&
-                Objects.equals(map, payload.map);
+                comparePayloadMap(map, payload.map);
+    }
+
+    private static boolean comparePayloadMap(final Map<String, Object> a, final Map<String, Object> b) {
```

Comment:
Note: this has been added to support force payload and serialization as the code base expects Arrays but the when deserializing and serializing payloads with maps Arrays of Objects get turned into Collections/Arraylists.